libs/core/langchain_core/runnables/history.py

```diff
@@ -483,6 +499,23 @@ def _exit_history(self, run: Run, config: RunnableConfig) -> None:
         output_messages = self._get_output_messages(output_val)
         hist.add_messages(input_messages + output_messages)
 
+    async def _aexit_history(self, run: Run, config: RunnableConfig) -> None:
```

Comment:
@mackong would you mind unit testing code to cover the async path?

Reply:
@eyurtsev unit testing added, but it will be failed caused by unrelated issue. 

I have create a PR #23214 which fix the issue, please review #23214 first.

Reply:
@eyurtsev now added unit tests passed